kafka-client/src/main/java/dev/responsive/kafka/api/config/ResponsiveConfig.java

```diff
@@ -204,13 +204,23 @@ public class ResponsiveConfig extends AbstractConfig {
   private static final String SUBPARTITION_HASHER_DOC = "Hasher to use for sub-partitioning.";
   private static final Class<?> SUBPARTITION_HASHER_DEFAULT = Murmur3Hasher.class;
 
+
+  // ------------------ Async processing configurations ----------------------
+
   public static final String ASYNC_THREAD_POOL_SIZE_CONFIG = "responsive.async.thread.pool.size";
   private static final int ASYNC_THREAD_POOL_SIZE_DEFAULT = 0;
   private static final String ASYNC_THREAD_POOL_SIZE_DOC = "The number of async processing threads to "
       + "start up for each StreamThread in this app. Setting this to 0 (the default) means async processing "
       + "will not be enabled. Setting this to a positive integer will enable async processing, but only if "
       + "there is at least one AsyncProcessor in the topology. See javadocs for AsyncProcessorSupplier for details.";
 
+  public static final String ASYNC_SCHEDULING_QUEUE_SIZE_CONFIG = "responsive.async.scheduling.queue.size";
+  private static final long ASYNC_SCHEDULING_QUEUE_SIZE_DEFAULT = 1L;
```

Comment:
why a default of 1 here? seems like it should be higher

Reply:
whoops, didn't mean to commit this. I agree

That said, I'm not really sure what a good default would be? I had it set to 100_000_000L initially as an arbitrary large-but-not-infinite value. I figured we could play around with it a bit when we have the perf testing set up. But idk -- thoughts?

Reply:
if we set it too high we'd risk buffering up so much we miss a poll right? I think something pretty small is actually just fine - like some multiple of the number of async threads.